repotracker/repotracker.go

```diff
@@ -825,15 +844,24 @@ func createVersionItems(ctx context.Context, v *model.Version, ref *model.Projec
 				"insert_err": err.Error(),
 				"abort_err":  abortErr.Error(),
 			})
-			return errors.Wrapf(err, "error inserting version %s", v.Id)
+			return false, errors.Wrapf(err, "error inserting version %s", v.Id)
 		}
 		err = sessCtx.CommitTransaction(sessCtx)
 		if err != nil {
+			cmdErr, isCmdErr := err.(mongo.CommandError)
+			if isCmdErr && cmdErr.HasErrorLabel(command.TransientTransactionError) {
```

Comment:
errors.cause() 

maybe factor this out to a function, also above.

Reply:
done